docs/source/Mapping.rst

```diff
@@ -0,0 +1,152 @@
+	                        ┌───┐└─┬─┘├───┤└─┬─┘┌───┐└╥┘          ┌─┐
+	q_2 -> 1 ────────────■──┤ X ├──■──┤ T ├──■──┤ X ├─╫───■───────┤M├──────
+	         ┌───┐     ┌─┴─┐└─┬─┘┌───┐└───┘     └─┬─┘ ║ ┌─┴─┐     └╥┘┌─┐
+	q_0 -> 2 ┤ H ├──■──┤ X ├──■──┤ T ├────────────■───╫─┤ X ├──■───╫─┤M├───
+	         └───┘┌─┴─┐├───┤     └───┘                ║ └───┘┌─┴─┐ ║ └╥┘┌─┐
+	q_1 -> 3 ─────┤ X ├┤ T ├──────────────────────────╫──────┤ X ├─╫──╫─┤M├
+	              └───┘└───┘                          ║      └───┘ ║  ║ └╥┘
+	    c: 4/═════════════════════════════════════════╩════════════╩══╩══╩═
+	                                                  3            2  0  1
+	Additional gates: 2
+	Additional SWAPs: 2
+        Runtime:          0.015844
+
+Not only does this solution yield less SWAPs, by default the :code:`compile` method optimizes SWAP gates in a post-processing step.
+
+Heuristic Mapping
+#################
+
+Finding an optimal mapping for a quantum circuit is an NP-hard problem. *QMAP* employs the `SMT solver Z3 <https://github.com/Z3Prover/z3>`_ to encode and solve the problem of finding an optimal mapping as a Boolean formula. This, of course, brings scalablity issues along with it.
+
+The *heuristic mapper* implemented in *QMAP* provides an alternative approach to quantum circuit mapping that trades off quality of the mapping result against runtime. Using the heuristic mapper works completely analogous to the exact mapper.
+
+    .. code-block:: python3
+
+        qc_mapped, res = qmap.compile(qc, arch, method="heuristic")
+
+        print(qc_mapped.draw(fold=-1))
+
+        print("Additional gates: %d" % res.json()["statistics"]["additional_gates"])
+        print("Additional SWAPs: %d" % res.json()["statistics"]["swaps"])
+        print("Runtime:          %f" % res.json()["statistics"]["mapping_time"])
+
+    .. code-block:: console
+
+	         ┌───┐┌───┐     ┌───┐                              ┌───┐   ┌─┐
+	q_0 -> 0 ┤ H ├┤ X ├──■──┤ T ├──────────────────────────────┤ X ├───┤M├
+	         └───┘└─┬─┘┌─┴─┐├───┤     ┌───┐               ┌───┐└─┬─┘┌─┐└╥┘
+	q_1 -> 1 ───────■──┤ X ├┤ X ├──■──┤ T ├────────────■──┤ X ├──■──┤M├─╫─
+	                   └───┘└─┬─┘┌─┴─┐└───┘┌───┐     ┌─┴─┐└─┬─┘ ┌─┐ └╥┘ ║
+	q_2 -> 2 ─────────────────■──┤ X ├──■──┤ T ├──■──┤ X ├──■───┤M├──╫──╫─
+	                             └───┘┌─┴─┐├───┤┌─┴─┐└┬─┬┘      └╥┘  ║  ║
+	q_3 -> 3 ─────────────────────────┤ X ├┤ T ├┤ X ├─┤M├────────╫───╫──╫─
+	                                  └───┘└───┘└───┘ └╥┘        ║   ║  ║
+	    c: 4/══════════════════════════════════════════╩═════════╩═══╩══╩═
+	                                                   3         2   0  1
+	Additional gates: 3
+	Additional SWAPs: 3
+        Runtime:          0.000065
+
+While this solution is not optimal anymore it only requires one more gate and even for such a small example the heuristic mapper is orders of magnitudes faster than the exact mapper.
```

Comment:
Also here a subsection about using teleportation in the heuristic mapper and referring to the respective paper would make sense. As above, if you feel comfortable to write such a section, go for it. If not, let's create an issue for that and tag Stefan in it, as he is the author of that paper.

Reply:
Same as above, gonna make it an issue.

Reply:
Sounds good. I'll resolve once the issues are created.

Reply:
Created the issue.